language/move-compiler/src/parser/lexer.rs

```diff
@@ -638,3 +637,68 @@ fn get_name_token(name: &str) -> Tok {
         _ => Tok::Identifier,
     }
 }
+
+// Trim the whitespace characters, include: space, tab, lf(\n) and crlf(\r\n).
+fn trim_whitespace(text: &str) -> &str {
+    let mut pos = 0;
+    let mut iter = text.chars();
+
+    while let Some(chr) = iter.next() {
+        match chr {
+            ' ' | '\t' | '\n' => pos += 1,
+            '\r' => {
+                let nch = iter.next();
+                if nch.is_some() && nch.unwrap() == '\n' {
+                    pos += 2;
+                    continue;
+                }
+
+                break;
+            }
+            _ => break,
+        };
+    }
+
+    &text[pos..]
+}
+
+#[cfg(test)]
+mod tests {
+    use super::trim_whitespace;
+
+    #[test]
+    fn test_trim_whitespace() {
```

Comment:
Need a test case for single `\r` which not be trimmed.

Reply:
OK,I will add a test case